src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectCapabilityResolver.cs

```diff
@@ -26,6 +27,8 @@ internal sealed class ProjectCapabilityResolver : IProjectCapabilityResolver, ID
     private readonly JoinableTaskFactory _jtf;
     private readonly CancellationTokenSource _disposeTokenSource;
 
+    private ImmutableDictionary<(string, string), bool> _cachedCapabilities = ImmutableDictionary<(string, string), bool>.Empty;
```

Comment:
This should have an OS-specific path comparer for key.Item1, right?

Reply:
This is VS only, so Windows only, and since false negatives are explicitly okay, I didn't think it was necessary.

Reply:
Windows-only would mean you should have an ignore case string comparer rather than the default, which is current culture.

Reply:
IMO saying that false negatives are explicitly okay is a teensy bit of an overstatement. As designed, the `TryGetCachedCapabilityMatch` method implies that it will return the match (true or false) if it is cached and will return false if it is not cached. However, it doesn't explicitly state that we played fast-and-loose with file paths so that it will return `false` when the value is actually cached because the path was passed with different casing.